WordPress/Classes/ViewRelated/Site Creation/Shared/ErrorStates/ErrorStateView.swift

```diff
@@ -0,0 +1,278 @@
+
+import UIKit
+
+import Gridicons
+import WordPressAuthenticator
+
+// MARK: ErrorStateView
+
+/// This view is intended for use as the root view of `ErrorStateViewController`.
+/// It manages the presentation of the various error states that can arise during the process of site creation.
+///
+final class ErrorStateView: UIView {
+
+    // MARK: Properties
+
+    /// A collection of parameters uses for animation & layout of the view.
+    private struct Parameters {
+        static let dismissalDimension           = CGFloat(16)
+        static let dismissalInsetScaleFactor    = CGFloat(0.05)
+        static let iPadWidth                    = CGFloat(512)
+        static let iPhoneWidthScaleFactor       = CGFloat(0.79)
+        static let retryTopPadding              = CGFloat(6)
+        static let stackViewSpacing             = CGFloat(10)
+        static let supportTopInset              = CGFloat(26)
+    }
+
+    /// This informs constraints applied to the view.
+    var preferredWidth: CGFloat {
+        let preferredWidth: CGFloat
+
+        if WPDeviceIdentification.isiPad() {
+            preferredWidth = Parameters.iPadWidth
+        } else {
+            let screenBounds = UIScreen.main.bounds
+            preferredWidth = screenBounds.width * Parameters.iPhoneWidthScaleFactor
+        }
+
+        return preferredWidth
+    }
+
+    /// The configuration of the error state view to apply.
+    private let configuration: ErrorStateViewConfiguration
+
+    /// This represents a "Dismiss" image view. It is visible when the configuration specifies a handler.
+    private(set) var dismissalImageView: UIImageView?
+
+    /// This represents the "headline" of the error message presented to the user.
+    private(set) var titleLabel: UILabel
+
+    /// This represents the "subtitle" of the error message presented to the user.
+    private(set) var subtitleLabel: UILabel?
+
+    /// This represents the "Retry" button associated with the error presentation. Visible when a handler is set.
+    private(set) var retryButton: UIButton?
+
+    /// This represents a container to add additional padding to the "Retry" button.
+    private(set) var retryContainerView: UIView?
+
+    /// This represents the "Contact support" of the error message presented to the user. Visible when a handler is set.
+    private(set) var contactSupportLabel: UILabel?
+
+    /// The stack view manages the central content of the stack view
+    private(set) var contentStackView: UIStackView
+
+    // MARK: ErrorStateView
+
+    /// The designated initializer.
+    ///
+    /// - Parameter configuration: the configuration with which to instantiate this error state view.
+    init(with configuration: ErrorStateViewConfiguration) {
+        self.configuration = configuration
+
+        self.titleLabel = {
```

Comment:
Same as with other previous PRs, I find this idiom makes the initialiser difficult to read. To me, it makes sense to instantiate these properties using closures, but I think it would read better if it was done via private lazy vars

Reply:
Right - we did discuss this [previously](https://github.com/wordpress-mobile/WordPress-iOS/pull/10609/). 

To recap, I agree that this benefits readability, but unfortunately there are issues with `lazy` and concurrency. We're not accessing this view from multiple threads, but that's troublesome.